doc/test_messages_documentation.py

```diff
@@ -53,7 +53,13 @@ def __init__(self, test_file: Tuple[str, Path]) -> None:
         self._linter.config.persistent = 0
         checkers.initialize(self._linter)
 
-        config_file = next(config.find_default_config_files(), None)
+        # Check if this message has a custom configuration file (e.g. for enabling optional checkers).
+        # If not, use the default configuration.
+        config_file: Optional[Path]
```

Comment:
@DanielNoord I think we talked about this already before but what do you think about using [``FunctionalTestFile``](https://github.com/PyCQA/pylint/blob/main/pylint/testutils/functional/test_file.py#L46) instead here ? At some point we will encounter the same issue than for functional tests (i.e. handling min_version, max_version, some options...). We could do ``test_file.option_file`` instead of recreating the logic a little differently.

Reply:
That was actually my initial approach, but the problem is that ``FunctionalTestFile`` assumes the config file to have the same name as the ``.py`` file.
There is quite a good portion of code duplication also in the ``LintModuleTest`` class here. 

Besides the different folder structure, the tight coupling between the "original" ``LintModuleTest`` and ``test_functional.py`` is another problem.

Reply:
Yeah, I advise against trying to use `LintModuleTest` here. The idea is similar but because of the different folder structures it doesn't really work.

It's similar to your `Duck` vs. `PlasticDuck` example @Pierre-Sassoulas  😄